user/src/com/google/gwt/user/client/impl/WindowImpl.java

```diff
@@ -29,11 +29,30 @@ public native String getHash() /*-{
   public native String getQueryString() /*-{
     return $wnd.location.search;
   }-*/;
-  
-  public native void initWindowCloseHandler() /*-{
+
+  @Deprecated
+  public void initWindowCloseHandler() {
+    initWindowUnloadHandler();
+    initWindowBeforeUnloadHandler();
+  }
+
+  @Deprecated
+  public native void initWindowUnloadHandler() /*-{
+    var oldOnUnload = $wnd.onunload;
+
+    $wnd.onunload = $entry(function(evt) {
+      try {
+        @com.google.gwt.user.client.Window::onClosed()();
+      } finally {
+        oldOnUnload && oldOnUnload(evt);
+        $wnd.onunload = null;
```

Comment:
Is there a reason why you kept this reset to `null` but removed the ones for `onresize`, `onscroll` and `onbeforeunload`?

Reply:
Since unload is not guaranteed to be run anyway, there is no benefit for other events - it won't consistently be invoked.

It will at least consistently clean itself up though - I'm _fairly_ certain that this only exists for (long since fixed) memory leaks, but it seemed like just in case there was some other clean up code somewhere, we'd want to still do this for itself. It does seem reasonable to fully remove though - open to your advice here.